main.tf

```diff
@@ -21,24 +21,11 @@ resource "azurerm_storage_account" "funcsta" {
   tags                      ="${var.common_tags}"
 }
 
-resource "azurerm_app_service_plan" "funcasp" {
-  name                = "${local.app_service_plan_name}"
-  location            = "${var.location}"
-  resource_group_name = "${var.resource_group_name}"
-  kind                = "${lower(var.plan_type) == "consumption" ? "FunctionApp" : var.plan_settings["kind"]}"
-  sku {
-    tier     = "${lower(var.plan_type) == "consumption" ? "Dynamic" : "Standard"}"
-    size     = "${lower(var.plan_type) == "consumption" ? "Y1" : var.plan_settings["size"]}"
-    capacity = "${lower(var.plan_type) == "consumption" ? 0 : var.plan_settings["capacity"]}"
-  }
-  tags                      ="${var.common_tags}"
-}
-
 resource "azurerm_function_app" "funcapp" {
   name                      = "${var.function_app_name}"
   location                  = "${var.location}"
   resource_group_name       = "${var.resource_group_name}"
-  app_service_plan_id       = "${azurerm_app_service_plan.funcasp.id}"
+  app_service_plan_id       = "${var.asp_id}"
```

Comment:
i guess you need to define that asp_id in variables.tf and README update. 
There is an existing branch with all these changes if you want to use it.